crates/typst/src/foundations/float.rs

```diff
@@ -122,6 +122,7 @@ cast! {
     v: f64 => Self(v),
     v: bool => Self(v as i64 as f64),
     v: i64 => Self(v as f64),
+    v: Decimal => Self(f64::try_from(v).map_err(|_| eco_format!("invalid float: {}", v))?),
```

Comment:
I've looked at the implementation of `impl TryFrom<rust_decimal::Decimal> for f64`, used for this cast, and it doesn't seem to actually ever return an error, but I'm handling the error case anyway just in case. Never know when stuff might change under us...